test/libsolidity/syntaxTests/inline_array_with_invalid_utf8_sequence.sol

```diff
@@ -0,0 +1,7 @@
+contract C {
+        function f1() external pure returns (string[1] memory rdatas) {
+                rdatas = [hex'c0a80101'];
```

Comment:
By the way, I think that we really need syntax for bytes literals. Currently even something as simple as `bytes[2] memory b = ["a", "b"]` won't compile without explicit casts to `bytes()`. This will be an usability issue now that array literals will be getting more use, given that `bytes` is much more common than strings (which are mostly useless).

Reply:
I have already replaced inline_array_with_invalid_utf8_sequence with tests for explicit and implicit conversions. Let me know is you still see any gaps in test coverage.

The good news is that the syntax for bytes literals is a part of this PR. I know it is not that obvious, so let me explain. When it comes to the conversion from array literal to an array, the compiler tries to find a common type for all array elements. To do that it uses mobile types of the elements. The mobile type of the string literal is String Array. So, the conversion path is: string literal -> string array -> byte array. Of course it fails because implicit conversion between string array and byte array is blocked. With this PR changes, compiler no mores search for common type. Now it tries to convert each element of an array literal to an element of the array. So, for each element there is an attempt to convert from a string literal to a  string. That means we have UTF8 validation for array literal elements and additionally get following conversions enabled:
`bytes[2] memory a = [hex'74000001', hex'c0a80101'];`
`bytes[2] memory a = ['foo', 'bar'];`.

Reply:
Btw, I have updated Changelog to highlight this change.

Reply:
Interesting. Yeah, with better implicit conversion maybe we don't need bytes literals that badly.

One more thing: Wouldn't it be better to extract the improved conversion to a separate PR? It would make it easier to review and it seems quite independent from adding support for dynamic array literals.

Reply:
I wish to do that, but afraid it might be not an easy task. I already tried decouple code related to UTF8 validation. But, it turned out that a majority of the code needs to be moved. Each of these topics spin around algorithm responsible for common type searching for  inline array elements. The algorithm is removed due to array literal handling requirements. Now we convert elements one by one directly to the target type. As a side effect, we get fixes for UTF8 validation and an implicit string literal conversion to bytes.